<a href="https://colab.research.google.com/github/AimoneCG/Proyecto-CoderHouse-Data-Science-I-Fundamentos-para-la-Ciencia-de-Datos/blob/main/ProyectoDS_Parte1_Aimone.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# **Primera Entrega** -  Cristian Aimone
# Data Science I : Fundamentos para la Ciencia de Datos
### Fecha: 17/10/2024


## Consigna:

### Visualizaciones en Python

1. Seleccionar uno de los datasets anterioriores *“Elección de Potenciales Datasets e importe con la librería Pandas”*.
2. Realizar un abstract (250/500 palabras) de acuerdo al dataset elegido.
3. Establecer las preguntas e hipótesis de interés sobre el dataset elegido.
    - Generar al menos dos preguntas y/o hipotesis.
4. Realizar gráficos con Matplotlib y con Seaborn.
    - 3 visualizaciones distintas con Matplotlib: 1 Univariado, 1 Bivariado, 1 libre elección.
    - 3 visualizaciones distintas con Seaborn: 1 Univariado, 1 Bivariado, 1 libre elección.
    - Generar una interpretación de los gráficos.
    - Debe usarse al menos un parametro adicional (grid, hue, etc) que enriquezca la legibilidad de los charts.

# Metadata


### Abstract

- El dataset elegido contiene una visión global de los estudiantes matriculados en diversas titulaciones de grado ofrecidas en una institución de enseñanza superior.
- Incluye datos demográficos, factores socioeconómicos e información sobre el rendimiento académico que pueden utilizarse para analizar los posibles factores predictivos del abandono y el éxito académico de los estudiantes.
- Este conjunto de datos contiene múltiples bases de datos disjuntas compuestas por información relevante disponible en el momento de la matriculación, como el modo de solicitud, el estado civil, el curso elegido, etc. Además, estos datos pueden utilizarse para estimar el rendimiento global de los estudiantes al final de cada semestre mediante la evaluación de las unidades curriculares acreditadas/matriculadas/evaluadas/aprobadas, así como sus respectivas calificaciones.
- Por último, disponemos de la tasa de desempleo, la tasa de inflación y el PIB de la región, que pueden llegar a ayudarnos a comprender mejor cómo influyen los factores económicos en las tasas de abandono escolar o en los resultados académicos.
- El análisis se realizará desde el punto de vista de una institución educativa cn el fin de obtener valiosa información sobre lo que motiva a los estudiantes a permanecer en la escuela o abandonar sus estudios para una amplia gama de disciplinas como agronomía, diseño, educación enfermería periodismo gestión servicio social o tecnologías


- Este conjunto de datos fue obtenido de : [https://www.kaggle.com/datasets/thedevastator/higher-education-predictors-of-student-retention] .

# Diccionario de Variables

- Marital status => Estado civil: Estado civil del alumno.
(Categórico)
- Application mode => Modo de solicitud: El método de solicitud utilizado por el estudiante. (Categórica)
- Application order => Orden de solicitud: El orden en que el estudiante presentó la solicitud. (Numérico)
- Course => Curso: Curso realizado por el estudiante. (Categórico)
- Daytime/evening attendance => Asistencia diurna/noche: Si el alumno asiste a clase durante el día o por la noche. (Categórico)
- Previous qualification => Titulación previa: La titulación obtenida por el estudiante antes de matricularse en la enseñanza superior. (Categórica)
- Nacionality => Nacionalidad: Nacionalidad del estudiante. (Categórica)
- Mother's qualification => Titulación de la madre: Titulación de la madre del estudiante. (Categórica)
- Father's qualification => Titulación del padre: Titulación del padre del alumno. (Categórica)
- Mother's occupation => Profesión de la madre: Profesión de la madre del estudiante. (Categórica)
- Father's occupation => Profesión del padre: Profesión del padre del alumno. (Categórica)
- Displaced => Desplazado: Si el estudiante es una persona desplazada. (Categórica)
- Educational special needs => Necesidades educativas especiales: Si el alumno tiene necesidades educativas especiales. (Categórico)
- Debtor => Deudor: Si el estudiante es deudor. (Categórico)
- Tuition fees up to date => Pago al día: Si las colegiaturas del estudiante están al día. (Categórico)
- Gender => Género: Genero o sexo del estudiante. (Categórico)
- Scholarship holder => Becario: Si el estudiante es becario. (Categórico)
- Age at enrollment => Edad de inscripción: Edad del estudiante en el momento de la inscripción. (Numérico)
- International => Internacional: Si el estudiante es internacional. (Categórico)
- Curricular units 1st sem (credited) => Unidades curriculares 1er sem (acreditadas): El número de unidades curriculares acreditadas por el estudiante en el primer semestre. (Numérico)
- Curricular units 1st sem (enrolled) => Unidades curriculares 1er sem (matriculadas): Número de unidades curriculares matriculadas por el alumno en el primer semestre. (Numérico)
- Curricular units 1st sem (evaluations) => Unidades curriculares 1er sem (evaluaciones): Número de unidades curriculares evaluadas por el alumno en el primer semestre. (Numérico)
- Curricular units 1st sem (approved) => Unidades curriculares 1er sem (aprobadas): Número de unidades curriculares aprobadas por el alumno en el primer semestre. (Numérico)

#Importación de librerias

In [1]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns

#Importación del dataset desde github

In [ ]:
url = "https://raw.githubusercontent.com/AimoneCG/Proyecto-CoderHouse-Data-Science-I-Fundamentos-para-la-Ciencia-de-Datos/main/dataset.csv"

df = pd.read_csv(url)

In [ ]:
df.head()

Marital status  Application mode  Application order  Course  \
0               1                 8                  5       2   
1               1                 6                  1      11   
2               1                 1                  5       5   
3               1                 8                  2      15   
4               2                12                  1       3   

   Daytime/evening attendance  Previous qualification  Nacionality  \
0                           1                       1            1   
1                           1                       1            1   
2                           1                       1            1   
3                           1                       1            1   
4                           0                       1            1   

   Mother's qualification  Father's qualification  Mother's occupation  ...  \
0                      13                      10                    6  ...   
1                       1                       3                    4  ...   
2                      22                      27                   10  ...   
3                      23                      27                    6  ...   
4                      22                      28                   10  ...   

   Curricular units 2nd sem (credited)  Curricular units 2nd sem (enrolled)  \
0                                    0                                    0   
1                                    0                                    6   
2                                    0                                    6   
3                                    0                                    6   
4                                    0                                    6   

   Curricular units 2nd sem (evaluations)  \
0                                       0   
1                                       6   
2                                       0   
3                                      10   
4                                       6   

   Curricular units 2nd sem (approved)  Curricular units 2nd sem (grade)  \
0                                    0                          0.000000   
1                                    6                         13.666667   
2                                    0                          0.000000   
3                                    5                         12.400000   
4                                    6                         13.000000   

   Curricular units 2nd sem (without evaluations)  Unemployment rate  \
0                                               0               10.8   
1                                               0               13.9   
2                                               0               10.8   
3                                               0                9.4   
4                                               0               13.9   

   Inflation rate   GDP    Target  
0             1.4  1.74   Dropout  
1            -0.3  0.79  Graduate  
2             1.4  1.74   Dropout  
3            -0.8 -3.12  Graduate  
4            -0.3  0.79  Graduate  

[5 rows x 35 columns]

In [ ]:
df.shape

(4424, 35)

El dataset contiene 4424 filas y 35 columnas.
Se puede apreciar que contiene tanto variables categóricas como númericas

In [ ]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 4424 entries, 0 to 4423
Data columns (total 35 columns):
 #   Column                                          Non-Null Count  Dtype  
---  ------                                          --------------  -----  
 0   Marital status                                  4424 non-null   int64  
 1   Application mode                                4424 non-null   int64  
 2   Application order                               4424 non-null   int64  
 3   Course                                          4424 non-null   int64  
 4   Daytime/evening attendance                      4424 non-null   int64  
 5   Previous qualification                          4424 non-null   int64  
 6   Nacionality                                     4424 non-null   int64  
 7   Mother's qualification                          4424 non-null   int64  
 8   Father's qualification                          4424 non-null   int64  
 9   Mother's occupation                      

El dataset no contiene valores nulos

In [ ]:
df.describe()

Marital status  Application mode  Application order       Course  \
count     4424.000000       4424.000000        4424.000000  4424.000000   
mean         1.178571          6.886980           1.727848     9.899186   
std          0.605747          5.298964           1.313793     4.331792   
min          1.000000          1.000000           0.000000     1.000000   
25%          1.000000          1.000000           1.000000     6.000000   
50%          1.000000          8.000000           1.000000    10.000000   
75%          1.000000         12.000000           2.000000    13.000000   
max          6.000000         18.000000           9.000000    17.000000   

       Daytime/evening attendance  Previous qualification  Nacionality  \
count                 4424.000000             4424.000000  4424.000000   
mean                     0.890823                2.531420     1.254521   
std                      0.311897                3.963707     1.748447   
min                      0.000000                1.000000     1.000000   
25%                      1.000000                1.000000     1.000000   
50%                      1.000000                1.000000     1.000000   
75%                      1.000000                1.000000     1.000000   
max                      1.000000               17.000000    21.000000   

       Mother's qualification  Father's qualification  Mother's occupation  \
count             4424.000000             4424.000000          4424.000000   
mean                12.322107               16.455244             7.317812   
std                  9.026251               11.044800             3.997828   
min                  1.000000                1.000000             1.000000   
25%                  2.000000                3.000000             5.000000   
50%                 13.000000               14.000000             6.000000   
75%                 22.000000               27.000000            10.000000   
max                 29.000000               34.000000            32.000000   

       ...  Curricular units 1st sem (without evaluations)  \
count  ...                                     4424.000000   
mean   ...                                        0.137658   
std    ...                                        0.690880   
min    ...                                        0.000000   
25%    ...                                        0.000000   
50%    ...                                        0.000000   
75%    ...                                        0.000000   
max    ...                                       12.000000   

       Curricular units 2nd sem (credited)  \
count                          4424.000000   
mean                              0.541817   
std                               1.918546   
min                               0.000000   
25%                               0.000000   
50%                               0.000000   
75%                               0.000000   
max                              19.000000   

       Curricular units 2nd sem (enrolled)  \
count                          4424.000000   
mean                              6.232143   
std                               2.195951   
min                               0.000000   
25%                               5.000000   
50%                               6.000000   
75%                               7.000000   
max                              23.000000   

       Curricular units 2nd sem (evaluations)  \
count                             4424.000000   
mean                                 8.063291   
std                                  3.947951   
min                                  0.000000   
25%                                  6.000000   
50%                                  8.000000   
75%                                 10.000000   
max                                 33.000000   

       Curricular units 2nd sem (approved)  Curricular units 2nd sem (grade)  \
count                          4424.000000        

# EDA - Análisis exploratorio de los datos

#Gráficos

###Hipótesis